nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/AbstractComponentNode.java

```diff
@@ -1084,12 +1092,29 @@ public void onParametersModified(final Map<String, ParameterUpdate> updatedParam
         }
 
         // If this component is affected by the Parameter change, we need to re-validate
+        componentAffected |= isConfigurationParameterModified(updatedParameters);
         if (componentAffected) {
             logger.debug("Configuration of {} changed due to an update to Parameter Context. Resetting validation state", this);
             resetValidationState();
         }
     }
 
+    protected void increaseReferenceCounts(final String parameterName) {
+        parameterReferenceCounts.merge(parameterName, 1, (a, b) -> a == -1 ? null : a + b);
+    }
+
+    protected void decreaseReferenceCounts(final String parameterName) {
+        parameterReferenceCounts.merge(parameterName, -1, (a, b) -> a == 1 ? null : a + b);
+    }
+
+    /**
+     * Verifies whether any referenced configuration parameter has been updated and its effective value has changed.
+     * If yes, the component needs to be re-validated.
+     * @param updatedParameters updated parameters
+     * @return true if the component requires re-validation
+     **/
+    public abstract boolean isConfigurationParameterModified(final Map<String, ParameterUpdate> updatedParameters);
```

Comment:
I'm not sure exactly what a "configuration parameter" is - i presume here we mean a parameter that is used within the configuration, other than properties, such as the Scheduling Period.  But not sure that this comes across from the code. Also not entirely sure that this is something that we should distinguish between - parameters that are used by properties and parameters that are used elsewhere. Perhaps this would call for a slight refactoring elsewhere but not sure that we should distinguish between the two. Should we just have a `Set<String> getReferencedParameterNames()` on ComponentNode? Would that give us everything that we need?

Reply:
This method has been removed.